src/board/helpers.mjs

```diff
@@ -0,0 +1,114 @@
+      newRow[rowIndex] = color;
+    });
+    result[gridIndex] = newRow;
+  });
+  return result;
+}
+
+function flattenGrid(grid) {
+  const result = [];
+  grid.forEach((row) => {
+    row.forEach((color) => {
+      result.push(color);
+    });
+  });
+  return result;
+}
+
+function shuffle(arr) { // Fisher-Yates shuffle
+  for (let i = arr.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1));
+    // TODO: ask about this
+    // eslint-disable-next-line no-param-reassign
+    [arr[i], arr[j]] = [arr[j], arr[i]];
+  }
+  return arr;
+}
+
+function flatToMultiDimensional(flatGrid) {
+  // square root of the size of the grid will tell us how many rows/columns
+  const rowSize = Math.sqrt(flatGrid.length);
+
+  // create the empty multidimensional grid
+  const multiDimensionalGrid = new Array(rowSize);
+
+  // why can I not turn this into a forEach or map
+  for (let i = 0; i < multiDimensionalGrid.length; i += 1) {
+    multiDimensionalGrid[i] = new Array(rowSize);
+  }
+
+  // use these variables to keep track of where in our md array we are as we iterate
+  let outerIndex = 0;
+  let innerIndex = 0;
+  // now iterate through the flatGrid and put all its items into the md grid
+  for (let i = 0; i < flatGrid.length; i += 1) {
+    multiDimensionalGrid[outerIndex][innerIndex] = flatGrid[i];
+    innerIndex += 1;
+    if (innerIndex % rowSize === 0) {
+      outerIndex += 1;
+      innerIndex = 0;
+    }
+  }
+
+  // but why is this angry about a lack of return statement
+  // flatGrid.map((color) => {
+  //   multiDimensionalGrid[outerIndex][innerIndex] = color;
+  //   innerIndex++;
+  //   if (innerIndex % 7 === 0) {
+  //     outerIndex++;
+  //     innerIndex = 0;
+  //   }
+  // })
+  return multiDimensionalGrid;
+}
+
+function shuffleGrid(grid) {
+  // shuffles grid but replaces the four corners back in their correct places
+  const gridSize = grid.length;
+  const flatGrid = flattenGrid(grid);
+  // create an object of the 4 corners
+  // where for each there is an array where the first index(0) is the index where we can locate
+  // the second index (1) is the value at that index
+  const corners = {
+    topleft: [0, flatGrid[0]],
+    topRight: [gridSize - 1, flatGrid[gridSize - 1]],
+    bottomLeft: [flatGrid.length - gridSize, flatGrid[flatGrid.length - gridSize]],
+    bottomRight: [flatGrid.length - 1, flatGrid[flatGrid.length - 1]],
+  };
+  const shuffledFlatGrid = shuffle(flatGrid);
+
+  for (const position in corners) {
+    const cornerIndex = corners[position][0];
+    const color = corners[position][1];
+    for (let i = 0; i < shuffledFlatGrid.length; i += 1) {
+      if (flatGrid[i] === color) {
+        // save the color that's at the corner index
+        const currentColor = flatGrid[cornerIndex];
+        // put the color back at the corner index
+        flatGrid[cornerIndex] = color;
+        // replace the color at the current index with the color that was originally there
```

Comment:
this file seems to have a lot of code comments, i think those can be really helpful but when there are so many it makes the code a little less readable because there is so much going on with code comments and the actual code, id suggest trying to only write comments when the code cant communicate the why of something rather than what it does. 

If you need comments to explain what something does you might need to rename some variables/functions, or break things up a bit.